### PR TITLE
Implement ConnectorResource deletion

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces enhanced deletion capabilities to `ConnectorResource`, tailored specifically to each `ConnectorProvider`. The goal is to consolidate the deletion process for each connector provider within a centralized place, ensuring that transactions remain confined to the resource layer.

## Risk

Main risk is around connector deletion, safe to rollback.

### CODE REVIEW

File: connectors/src/connectors/confluence/index.ts [hunk 1]
```diff
@@ -235,40 +235,17 @@ export async function cleanupConfluenceConnector(
...
-    ]);
-
-    const nangoRes = await nangoDeleteConnection(
-      connector.connectionId,
-      getRequiredNangoConfluenceConnectorId()
-    );
-    if (nangoRes.isErr()) {
-      throw nangoRes.error;
-    }
+  await connector.delete();
```
Comment (spolu) [inline connectors/src/connectors/confluence/index.ts:238]: Can this fail? Compared to github error handling we probably want to align the two?

**Review by fontanierh — APPROVED** [review]
quite nice

File: connectors/src/connectors/confluence/index.ts (hunk 1 above)
Comment (spolu) [reply connectors/src/connectors/confluence/index.ts:238]: I thiiiiink connector deletion should come after nango deletion?

**Review by spolu — APPROVED** [review]
LGTM

Left a few comments

File: connectors/src/connectors/confluence/index.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/src/connectors/confluence/index.ts]: We always have different error handling in all the cleanup functions, let me take a stab at those and align them on the GitHub example.

Comment (flvndvd) [reply connectors/src/connectors/confluence/index.ts]: Fix in https://github.com/dust-tt/dust/pull/3812/commits/620a58d5ceaf66f2b7e1a7f0738e0dbbed5280ee.

**Review by spolu — APPROVED** [review]
👍

**Pull request merged**